docs: add docs for serialization of DocList (#1350)
docarray/docarray: docs/user_guide/sending/ser/send_doclist.md

# DocList
When sending or storing [`DocList`][docarray.array.doc_list.doc_list.DocList], you need to use serialization. [DocList][docarray.array.doc_list.doc_list.DocList] supports multiple ways to serialize the data.

## JSON

> You can use [`to_json()`][docarray.array.doc_list.doc_list.DocList.to_json] and [`from_json()`][docarray.array.doc_list.doc_list.DocList.from_json] to serialize and deserialize a [DocList][docarray.array.doc_list.doc_list.DocList].

Reviewer: ```suggestion
You can use [`to_json()`][docarray.array.doc_list.doc_list.DocList.to_json] and [`from_json()`][docarray.array.doc_list.doc_list.DocList.from_json] to serialize and deserialize a [DocList][docarray.array.doc_list.doc_list.DocList]:
```

Please use `:` before code samples. Fix throughout PR

## Base64
When transferring over the network, you can choose `Base64` format to serialize the [`DocList`][docarray.array.doc_list.doc_list.DocList].
Serializing a [DocList][docarray.array.doc_list.doc_list.DocList] in Base64 supports both `pickle` and `protobuf` protocols. Besides, you can choose different compression methods.

To serialize a [DocList][docarray.array.doc_list.doc_list.DocList] in Base64, you can use [`to_base64()`][docarray.array.doc_list.doc_list.DocList.to_base64]  and [`from_base64()`][docarray.array.doc_list.doc_list.DocList.from_protobuf] to serialize and deserialize a [DocList][docarray.array.doc_list.doc_list.from_base64].

We support multiple compression methods. (namely : `lz4`, `bz2`, `lzma`, `zlib`, `gzip`)

```python
from docarray import BaseDoc, DocList

class SimpleDoc(BaseDoc):
    text: str

dl = DocList[SimpleDoc]([SimpleDoc(text=f'doc {i}') for i in range(2)])

base64_repr_dl = dl.to_base64(compress=None, protocol='pickle')

dl_from_base64 = DocList[SimpleDoc].from_base64(
    base64_repr_dl, compress=None, protocol='pickle'
)
```

## Binary

> Similar as in `Base64` serialization, `Binary` serialization also supports different protocols and compression methods.

Reviewer: ```suggestion
Similar to `Base64` serialization, `Binary` serialization also supports different protocols and compression methods.
```
